warehouse/slave.go

```diff
@@ -462,19 +462,36 @@ func processStagingFile(job PayloadT) (loadFileIDs []int64, err error) {
 
 func claimAndProcess(workerIdx int, slaveID string) {
 	pkgLogger.Debugf("[WH]: Attempting to claim job by slave worker-%v-%v", workerIdx, slaveID)
+
+	freeWorker := func() {
+		slaveWorkerRoutineBusy[workerIdx-1] = false
```

Comment:
Instead of using workerIdx-1, can we change the for loop in `setupSlave()` to iterate from 0 -> noOfSlaveWorkerRoutines - 1